src/services/verify/fragments.tsx

```diff
@@ -1,23 +1,24 @@
-import { isValid, VerificationFragment } from "@govtechsg/oa-verify";
-const getFirstFragmentFor = (fragments: VerificationFragment[], name: string) =>
+import { isValid, VerificationFragment, AllVerificationFragment } from "@govtechsg/oa-verify";
+
+const getFirstFragmentFor = (fragments: AllVerificationFragment[], name: string) =>
   fragments.filter((status) => status.name === name)[0];
 
 const revokeFragmentName = "OpenAttestationEthereumDocumentStoreRevoked";
-export const getNotRevokeFragment = (fragments: VerificationFragment[]): VerificationFragment<any>[] =>
+export const getNotRevokeFragment = (fragments: VerificationFragment[]): VerificationFragment[] =>
```

Comment:
think pretty much all in this file can be replaced with utilities from oa-verify
https://github.com/Open-Attestation/oa-verify/blob/f10e87e9c2515059e1c5eb661d12655111695170/README.md#L213-L214

Reply:
i actually dont see the similiar functions in oa-verify, do u mean we need to move this logic to oa-verify?

Reply:
revoked/issued = document status
hashValid = integrity

so just modify interpretFragments to directly use oa-verify utils to retrieve these fragments and decide on the pass/fail for those categories

Reply:
i understand that but the internal function are different so the result will be different, so for example, getRevokeFragment function uses a filter with 2 conditions, but in the oa-verify it is using .find for getFragmentByName and it return only 1 fragment. also, for the ```hashValid = isValid(fragments, ["DOCUMENT_INTEGRITY"]);``` if we want to have the same boolean we need to do something like ``` hashValid = isValid(getDocumentIntegrityFragments(fragments), ["DOCUMENT_INTEGRITY"])``` which i think is not necessary, can please advice on this. maybe can i schedule a one on one session with u?

Reply:
`hashValid = isValid(getDocumentIntegrityFragments(fragments), ["DOCUMENT_INTEGRITY"])` looks good to me actually, why do you feel its not necessary?
use the exported enum for "DOCUMENT_INTEGRITY" too, not sure if any necessity but there are also error codes to check against (maybe for revoked?)

using these utils will make upgrading to v3 much easier in the future

Reply:
i think its unnecessary because the code run filter twice, 1st in getDocumentIntegrityFragment and 2nd in isValid.
because we are still passing in the fragments. but if its ok with u then ill just add it in.
ok understand about the enum, will update it accordingly.

Reply:
Any way i have updated to use utils from oa-verify, let me know if its ok.

Reply:
ok it will be too much irrelevant changes to 'updating deps' to do what i had in mind in this PR, so please revert this and create a new story+pr to incorporate the same style of error checks as https://github.com/OpenCerts/opencerts-website/blob/dd10456fba5f29538e3ff1cdda54aa0f8dfc9976/src/services/fragment.ts#L6-L6 , and upstreaming these utils into open-attestation